packages/hub/src/hub.ts

```diff
@@ -373,6 +371,14 @@ export class Hub implements HubInterface {
       ...context,
     });
     if (scope) {
+      // End existing session if there's one
+      const session = scope.getSession && scope.getSession();
+      if (session) {
+        session.update({ status: SessionStatus.Exited });
+      }
```

Comment:
```suggestion
      const currentSession = scope.getSession && scope.getSession();
      if (currentSession) {
        currentSession.update({ status: SessionStatus.Exited });
      }
```

Just so it's more obvious that this is different from the already-declared `session`.